doc/designs/edns.md

```diff
@@ -0,0 +1,141 @@
+
+The following diagram represents a topology involving FreeIPA Server, Replica and Client:
+
+![edns-diagram](edns/FreeIPA-eDNS-version3.jpg)
+
+Note from the diagram that encryption is needed when queries are leaving the machines. When communication happens inside the host only, encryption is not necessary.
+
+FreeIPA currently relies on systemd-resolved as a local cache resolver, which is enabled by default. The new design involves disabling the systemd-resolved service and replacing it with the Unbound service. The client configuration relies exclusively on Unbound, with a DoT forwarder pointing to the DNS server. The FreeIPA server configuration consists of two main components: BIND (named) as an integrated DNS server, accepting both incoming unencrypted queries from localhost and incoming encrypted queries from external traffic, while relying on Unbound for handling outgoing external encrypted traffic. 
+
+The FreeIPA replica deployment depends on whether the DNS integrated service is deployed, distinguishing between two use cases: with and without DNS integrated service. A replica with DNS Integrated Service will mimic a client configuration: it will use Unbound with a DoT forwarder pointing to the DNS server. A replica without DNS Integrated Service will mimic a server configuration: it will use BIND for handling incoming unencrypted queries from localhost and encrypted queries from external sources, along with Unbound for outgoing encrypted traffic. 
+
+Finally, the trust chain is a critical component of setting up a DoT DNS channel as it enables the client to verify the identity of the DNS server and establish a secure and trusted connection for encrypted DNS communication. To enhance flexibility and accommodate varying deployment scenarios, we plan to introduce additional parameters that allow administrators to provide a certificate at FreeIPA installation time by allowing administrator to specify a custom certificate. The new parameter will be optional, giving administrators the choice to either provide a certificate or rely on custodia for certificate auto enrollment if no certificate is provided.
+
+## Implementation
+
+Most changes involve configuration updates. New installation options for encrypted DNS ensure the deployment and activation of DNS services to encrypt all outbound traffic.
+
+Address of our unbound server has to be set in /etc/resolv.conf. /etc/resolv.conf nameserver 127.0.0.1
```

Comment:
We will be using 127.0.0.53 or 127.0.0.53 (depends on whether we disable systemd-resolved completely or not).

Reply:
I think we should deploy the simplest configuration. I would opt for Unbound and BIND instead of systemd-resolved, Unbound, and BIND. Then, I would use 127.0.0.53 because it runs on port 53, avoiding conflicts with systemd-resolved if it is enabled later by mistake or manually.